vcd/resource_vcd_external_network_v2.go

```diff
@@ -0,0 +1,453 @@
+		return fmt.Errorf("could not get network data: %s", err)
+	}
+
+	extNet, err := govcd.CreateExternalNetworkV2(vcdClient.VCDClient, netType)
+	if err != nil {
+		return fmt.Errorf("error applying data: %s", err)
+	}
+
+	// Only store ID and leave all the rest to "READ"
+	d.SetId(extNet.ExternalNetwork.ID)
+
+	return resourceVcdExternalNetworkV2Read(d, meta)
+}
+
+func resourceVcdExternalNetworkV2Update(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] update network V2 creation initiated")
+
+	extNet, err := govcd.GetExternalNetworkV2ById(vcdClient.VCDClient, d.Id())
+	if err != nil {
+		return fmt.Errorf("could not find external network by ID '%s': %s", d.Id(), err)
+	}
+
+	netType, err := getExternalNetworkV2Type(vcdClient, d)
+	if err != nil {
+		return fmt.Errorf("could not get network data: %s", err)
+	}
+
+	netType.ID = extNet.ExternalNetwork.ID
+	extNet.ExternalNetwork = netType
+
+	_, err = extNet.Update()
+	if err != nil {
+		return fmt.Errorf("error updating external network: %s", err)
+	}
+
+	return resourceVcdExternalNetworkV2Read(d, meta)
+}
+
+func resourceVcdExternalNetworkV2Read(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] external network V2 read initiated")
+
+	extNet, err := govcd.GetExternalNetworkV2ById(vcdClient.VCDClient, d.Id())
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("could not find external network by ID '%s': %s", d.Id(), err)
+	}
+
+	return setExternalNetworkV2Data(d, extNet.ExternalNetwork)
+}
+
+func resourceVcdExternalNetworkV2Delete(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] external network V2 creation initiated")
+
+	extNet, err := govcd.GetExternalNetworkV2ById(vcdClient.VCDClient, d.Id())
+	if err != nil {
+		return fmt.Errorf("could not find external network by ID '%s': %s", d.Id(), err)
+	}
+
+	return extNet.Delete()
+}
+
+// resourceVcdExternalNetworkV2Import is responsible for importing the resource.
+// The d.ID() field as being passed from `terraform import _resource_name_ _the_id_string_ requires
+// a name based dot-formatted path to the object to lookup the object and sets the id of object.
+// `terraform import` automatically performs `refresh` operation which loads up all other fields.
+// For this resource, the import path is just the external network name.
+//
+// Example import path (id): externalNetworkName
+// Example import command:   terraform import vcd_external_network_v2.externalNetworkName externalNetworkName
+func resourceVcdExternalNetworkV2Import(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+
+	vcdClient := meta.(*VCDClient)
+
+	extNetRes, err := govcd.GetExternalNetworkV2ByName(vcdClient.VCDClient, d.Id())
+	if err != nil {
+		return nil, fmt.Errorf("error fetching external network details %s", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("error fetching external network v2 details %s", err)
```

Reply:
Good. Spotted a at least a few other errors like that. Fixed all.